bigtable/client/instance_admin_test.cc

```diff
@@ -488,8 +488,278 @@ TEST_F(InstanceAdminTest, CreateInstancePollReturnsFailure) {
+TEST_F(InstanceAdminTest, UpdateInstancePollReturnsFailure) {
+  using ::testing::_;
+  using ::testing::Invoke;
+
+  bigtable::InstanceAdmin tested(client_);
+  EXPECT_CALL(*client_, UpdateInstance(_, _, _))
+      .WillOnce(Invoke([](grpc::ClientContext*,
+                          btproto::PartialUpdateInstanceRequest const& request,
+                          google::longrunning::Operation* response) {
+        return grpc::Status::OK;
+      }));
+
+  EXPECT_CALL(*client_, GetOperation(_, _, _))
+      .WillOnce(Invoke([](grpc::ClientContext*,
+                          google::longrunning::GetOperationRequest const&,
+                          google::longrunning::Operation* operation) {
+        operation->set_done(false);
+        return grpc::Status::OK;
+      }))
+      .WillOnce(Invoke([](grpc::ClientContext*,
+                          google::longrunning::GetOperationRequest const&,
+                          google::longrunning::Operation* operation) {
+        operation->set_done(false);
+        return grpc::Status::OK;
+      }))
+      .WillOnce(
+          Invoke([](grpc::ClientContext*,
+                    google::longrunning::GetOperationRequest const& request,
+                    google::longrunning::Operation* operation) {
+            operation->set_done(true);
+            auto error = bigtable::internal::make_unique<google::rpc::Status>();
+            error->set_code(grpc::StatusCode::FAILED_PRECONDITION);
+            error->set_message("something is broken");
+            operation->set_allocated_error(error.release());
+            return grpc::Status::OK;
+          }));
+
+  btproto::Instance instance;
+  google::protobuf::FieldMask update_mask;
+  auto future = tested.UpdateInstance(&instance, &update_mask);
+  EXPECT_THROW(future.get(), bigtable::GRpcError);
+}
 #endif  // GOOGLE_CLOUD_CPP_HAVE_EXCEPTIONS
 
+/// @test Verify that `bigtable::InstanceAdmin::UpdateInstance` works.
+TEST_F(InstanceAdminTest, UpdateInstance) {
+  using ::testing::_;
+  using ::testing::Invoke;
+  bigtable::InstanceAdmin tested(client_);
+
+  EXPECT_CALL(*client_, UpdateInstance(_, _, _))
+      .WillOnce(Invoke([](grpc::ClientContext*,
+                          btproto::PartialUpdateInstanceRequest const& request,
+                          google::longrunning::Operation* response) {
+        auto const instance_name =
+            "projects/my-project/instances/test-instance";
+        EXPECT_EQ(instance_name, request.instance().name());
+        return grpc::Status::OK;
+      }));
+
+  std::string expected_text = R"(
+name: 'projects/my-project/instances/test-instance'
+display_name: 'foo bar'
+state: READY
+type: PRODUCTION
+labels: {
+  key: 'foo1'
+  value: 'bar1'
+}
+labels: {
+  key: 'foo2'
+  value: 'bar2'
+}
+)";
+  btproto::Instance expected;
+  ASSERT_TRUE(
+      google::protobuf::TextFormat::ParseFromString(expected_text, &expected));
+
+  std::string update_mask_text = R"(
+  paths: 'display_name'
```

Comment:
Let's be consistent about the indentation of these strings.  Elsewhere @mbrukman suggested indent 4 from the code, I like that, please replace this to:

```C++
  std::string update_mask_text = R"(
      paths: 'display_name'
      paths: 'labels'
  )";
```

Reply:
fixed